### ISSUE TITLE
Create a GitHub Actions to draft the release announcements

### ISSUE THREAD

**Issue by samccann** [ansible-community/ansible-build-data#351]
Since we've had some typos etc in recent release announcements for the Ansible package, the idea surfaced to create a GitHub Actions that would take the template announcements being created in https://github.com/ansible-community/ansible-build-data/pull/258 and fill in the correct details.

**Comment by samccann** [comment]
From a matrix chat:
Hey folks - 

@oraNod and I were chatting earlier about the possibility of using github actions to help create a clean release announcement (and maybe auto-post it to the news/newsletter category in the forum). It seems something that's ripe for automation and could simplify the release manager's task (especially as the announcement is the last thing on a long list of tasks).
So he can correct me where I'm getting fuzzy but if we had a generic template in the build data dir, we could use github actions and some variables to fill in the correct long-form version (core 2.16.1 and ansible 9.1.0 for example( and short form version (2.16 an d 9)

And also the other bits and bobs - correct link to pypi, the SHA and zip file locations etc etc.  Basically take human error out of it all and let automation create it.  From there, with some discourse API magic, it could be autoposted (much like the meeting logs are autoposted today).

**Comment by gotmax23** [comment]
I would suggest a script to generate this based on a Jinja2 template. It could interact with the Discourse API and/or sendmail to auto-post.

**Comment by oraNod** [comment]
@gotmax23 Yeah that's pretty much exactly what I mentioned to @samccann We can create a jinja template and a script to generate the announcements based on a few inputs.

**Comment by oraNod** [comment]
I guess I could just assign this one to myself and send a PR...

**Comment by oraNod** [comment]
@samccann Hey I've been hacking around with this a bit this evening. Still have some things to work out with posting to discourse but I pushed a couple changes to my fork that you can try out. It at least demonstrates what I had in mind about passing inputs to the action then generating an announcement from a standard template.

https://github.com/oraNod/ansible-build-data/actions/workflows/announce_major_release.yml

![image](https://github.com/ansible-community/ansible-build-data/assets/32749632/5ea5c1ca-2164-4884-9ea5-aac05ba492f7)

**Comment by samccann** [comment]
Hey looks good! I don't have the 'run workflow' option but I can see what the workflow is doing. I think the other input has to be the core version for sure. But maybe the better approach is to get  https://github.com/ansible-community/ansible-build-data/pull/258 merged and then figure out what other inputs there might be.
@anweshadas what do you think?

**Comment by oraNod** [comment]
Thanks. Getting #258 merged first seems like a good way to go. We can follow that up with the inputs and workflow to generate templates.

**Comment by anweshadas** [comment]
Yes, can we please merge [#258](https://github.com/ansible-community/ansible-build-data/pull/258).

**Comment by webknjaz** [comment]
@anweshadas could you address the review comments in #258?

**Comment by gotmax23** [comment]
Inspired by @oraNod's work, I started working on an antsibull-build subcommand to generate these announcements based on jinja2 templates in https://github.com/ansible-community/antsibull/pull/573. I would like to add functionality to open pre-filled mailto: and forum entry links with the generated announcement text as well.

**Comment by webknjaz** [comment]
> I would like to add functionality to open pre-filled mailto:

It should be possible to use https://docs.python.org/3/library/webbrowser.html#webbrowser.open for this. FYI.
